ballerina/soap_utils.bal

```diff
@@ -1,106 +1,198 @@
+
+public function validateTransportBindingPolicy(ClientConfig config) returns Error? {
+    if config.httpConfig.secureSocket is () {
+        wssec:InboundSecurityConfig|wssec:InboundSecurityConfig[] securityPolicy = config.inboundSecurity;
+        if securityPolicy is wssec:TransportBindingConfig {
+            return error Error(INVALID_PROTOCOL_ERROR);
+        } else if securityPolicy is wssec:InboundSecurityConfig[] {
+            foreach wssec:InboundSecurityConfig policy in securityPolicy {
+                if policy is wssec:TransportBindingConfig {
+                    return error Error(INVALID_PROTOCOL_ERROR);
+                }
+            }
+        }
+    }
+}
+
+public function retrieveHttpClientConfig(ClientConfig config) returns http:ClientConfiguration {
+    return {
+        httpVersion: config.httpConfig.httpVersion,
+        http1Settings: config.httpConfig.http1Settings,
+        http2Settings: config.httpConfig.http2Settings,
+        timeout: config.httpConfig.timeout,
+        poolConfig: config.httpConfig?.poolConfig,
+        auth: config.httpConfig?.auth,
+        retryConfig: config.httpConfig?.retryConfig,
+        responseLimits: config.httpConfig.responseLimits,
+        secureSocket: config.httpConfig?.secureSocket,
+        circuitBreaker: config.httpConfig?.circuitBreaker
+    };
+}
+
+public function applySecurityPolicies(wssec:InboundSecurityConfig|wssec:InboundSecurityConfig[] inboundSecurity,
+                                      xml envelope) returns xml|wssec:Error {
+    wssec:InboundSecurityConfig|wssec:InboundSecurityConfig[] securityPolicy = inboundSecurity;
+    xml securedEnvelope;
+    if securityPolicy is wssec:InboundSecurityConfig {
+        if securityPolicy is wssec:TimestampTokenConfig {
+            securedEnvelope = check wssec:applyTimestampToken(envelope, securityPolicy);
+        } else if securityPolicy is wssec:UsernameTokenConfig {
+            securedEnvelope = check wssec:applyUsernameToken(envelope, securityPolicy);
+        } else if securityPolicy is wssec:SymmetricBindingConfig {
+            securedEnvelope = check wssec:applySymmetricBinding(envelope, securityPolicy);
+        } else if securityPolicy is wssec:AsymmetricBindingConfig {
+            securedEnvelope = check wssec:applyAsymmetricBinding(envelope, securityPolicy);
+        } else {
+            securedEnvelope = envelope;
+        }
+    } else {
+        foreach wssec:InboundSecurityConfig policy in securityPolicy {
+            securedEnvelope = check applySecurityPolicies(policy, envelope);
+        }
+    }
+    return securedEnvelope;
+}
+
+public function applyOutboundConfig(wssec:OutboundSecurityConfig outboundSecurity, xml envelope)
+    returns xml|Error {
+    xmlns "http://schemas.xmlsoap.org/soap/envelope/" as soap;
+    xml soapEnvelope = envelope;
+    do {
+        wssec:EncryptionAlgorithm? encryptionAlgorithm = outboundSecurity.decryptionAlgorithm;
+        if encryptionAlgorithm !is () {

```

Comment:
```suggestion
        if encryptionAlgorithm is wssec:EncryptionAlgorithm {
```
shall we change other places too ?

Reply:
Fixed in [1de6750](https://github.com/ballerina-platform/module-ballerina-soap/pull/103/commits/1de6750f3a32d73e90b0ade50476774f2541fccf)